### PR TITLE
Release workflow update

### PR DESCRIPTION
**Pull request by bam241**
this will change:
- [x] CMakeFile will generate header for python and CPP to deal with  PYNE_VERSION number 
- [x] change the DAGMC target in CI Docker image to target DAGMC/`stable` tag (just been created)

### CODE REVIEW

**Comment by bam241** [comment]
the CI related to docker update can be found there: https://github.com/bam241/pyne/actions/runs/1254336958

**Comment by bam241** [comment]
this is ok on my side :)

File: CHANGELOG.rst [hunk 1]
```diff
@@ -10,6 +10,8 @@ Next Version
 **Change**
    * exclude docker related stuff from build_test (#1404 #1408)
    * move changelog test in its own workflow (#1404)
+   * Adds 2 templates header files (one for Python one for CPP) to store PYNE version number, cmake will automatically create proper header for python and cpp based on the PYNE VERSION number located in the root CMakeFile.txt (#1415)
```
Comment (pshriwise) [inline CHANGELOG.rst]:
```suggestion
   * Adds 2 templates header files (one for Python one for CPP) to store PYNE version number, cmake will automatically configure proper version files for python and cpp based on the PYNE VERSION number located in the root CMakeFile.txt (#1415)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,19 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
...
+
+# Set PYNE version
+set(PYNE_MAJOR_VERSION 0)
+set(PYNE_MINOR_VERSION 7)
+set(PYNE_PATCH_VERSION 5)
+set(PYNE_VERSION ${PYNE_MAJOR_VERSION}.${PYNE_MINOR_VERSION}.${PYNE_PATCH_VERSION})
+
+# Configure Pyne and cpp version headers
+configure_file(pyne/pyne_version.py.in ${CMAKE_CURRENT_SOURCE_DIR}/pyne/pyne_version.py)
+#install(FILES pyne/DAGMCConfig.cmake DESTINATION ${INSTALL_LIB_DIR}/cmake/)
```
Comment (pshriwise) [inline CMakeLists.txt]: Is this commented line necessary?

File: setup.py [hunk 3]
```diff
@@ -60,7 +60,9 @@
 def absexpanduser(x): return os.path.abspath(os.path.expanduser(x))
 
 
-VERSION = '0.7.1'
+# from pyne.pyne_version import PYNE_VERSION
```
Comment (pshriwise) [inline setup.py]: Looks like we can remove commented these lines in favor of those in `setup_sub.py`.

File: setup_sub.py [hunk 4]
```diff
@@ -23,7 +23,10 @@
 from distutils.core import setup
 
 
-VERSION = '0.7.1'
+from pyne.pyne_version import PYNE_VERSION
+VERSION = PYNE_VERSION
```
Comment (pshriwise) [inline setup_sub.py]: Looks like we can save a line and use `PYNE_VERSION` in the `setup_kwargs` dictionary below.

File: pyne/__init__.py [hunk 5]
```diff
@@ -2,15 +2,17 @@
...
 
-__version__ = '0.7.1'
 
 if os.name == 'nt':
     p = os.environ['PATH'].split(';')
     lib = os.path.join(os.path.split(__file__)[0], 'lib')
     os.environ['PATH'] = ";".join([lib] + p)
 
 try:
+    from .pyne_version import PYNE_VERSION
```
Comment (pshriwise) [inline pyne/__init__.py:12]: If we've set the `version` keyword in the `setup` call when installing PyNE, why do we need these lines?

**Review by pshriwise — CHANGES_REQUESTED** [review]
Thanks @bam241. Just a few comments/questions here.

File: pyne/__init__.py (hunk 5 above)
Comment (bam241) [reply pyne/__init__.py:12]: I don't know, I just kept everything as it was

Comment (pshriwise) [reply pyne/__init__.py:12]: Hrmm I see what you mean. That's odd to me though. Gonna try this out locally....

File: CMakeLists.txt [hunk 6]
```diff
@@ -11,6 +11,18 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 # Setup the project
 project(pyne)
 
+
+# Set PYNE version
+set(PYNE_MAJOR_VERSION 0)
+set(PYNE_MINOR_VERSION 7)
+set(PYNE_PATCH_VERSION 5)
```
Comment (pshriwise) [inline CMakeLists.txt:18]: It looks like the version before this PR was `0.7.1`. It seems to me we should keep it the same for now and do a release with this update later. Is that possible?

**Review by gonuke — APPROVED** [review]
Other than @pshriwise's comments, this LGTM

File: CMakeLists.txt (hunk 6 above)
Comment (bam241) [reply CMakeLists.txt:18]: the last version tagged on Github is 0.7.5 that's why I set it like that

Comment (gonuke) [reply CMakeLists.txt:18]: Oh yeah... that's right. We have advanced release without advancing the version number, which is one of the motivations for this PR

**Review by pshriwise — APPROVED** [review]
All convos have been resolved. Thanks for clarifying those issues @bam241 and @gonuke.

**Pull request merged**